components/motor/dimensionengineering/sabertooth.go

```diff
@@ -0,0 +1,556 @@
+	MaxRPM float64 `json:"max_rpm,omitempty"`
+
+	// The name of the encoder used for this motor
+	Encoder string `json:"encoder,omitempty"`
+
+	// The lowest power percentage to allow for this motor. This is used to prevent motor stalls and overheating. Default is 0.0
+	MinPowerPct float64 `json:"min_power_pct,omitempty"`
+
+	// The max power percentage to allow for this motor. Default is 0.0
+	MaxPowerPct float64 `json:"max_power_pct,omitempty"`
+
+	// The number of ticks per rotation of this motor from the encoder
+	TicksPerRotation int `json:"ticks_per_rotation,omitempty"`
+
+	// TestChan is a fake "serial" path for test use only
+	TestChan chan []byte `json:"-,omitempty"`
+}
+
+// Validate ensures all parts of the config are valid.
+func (cfg *Config) Validate(path string) error {
+	if cfg.SerialPath == "" {
+		return utils.NewConfigValidationFieldRequiredError(path, "serial_path")
+	}
+	if cfg.MotorChannel < 1 || cfg.MotorChannel > 2 {
+		return utils.NewConfigValidationError(path, errors.New("invalid channel, acceptable values are 1 and 2"))
+	}
+	if cfg.SerialAddress < 128 || cfg.SerialAddress > 135 {
+		return utils.NewConfigValidationError(path, errors.New("invalid address, acceptable values are 128 thru 135"))
+	}
+	if cfg.BaudRate != 2400 && cfg.BaudRate != 9600 && cfg.BaudRate != 19200 && cfg.BaudRate != 38400 && cfg.BaudRate != 115200 {
+		return utils.NewConfigValidationError(path, errors.New("invalid baud_rate, acceptable values are 2400, 9600, 19200, 38400, 115200"))
+	}
+	if cfg.MinPowerPct < 0.0 || cfg.MinPowerPct > cfg.MaxPowerPct {
+		return utils.NewConfigValidationError(path, errors.New("invalid min_power_pct, acceptable values are 0 to max_power_pct"))
+	}
+	if cfg.MaxPowerPct > 100.0 {
+		return utils.NewConfigValidationError(path, errors.New("invalid max_power_pct, acceptable values are min_power_pct to 100.0"))
+	}
+
+	return nil
+}
+
+func init() {
+	controllers = make(map[string]*controller)
+
+	_motor := registry.Component{
+		Constructor: func(ctx context.Context, _ registry.Dependencies, config config.Component, logger golog.Logger) (interface{}, error) {
+			conf, ok := config.ConvertedAttributes.(*Config)
+			if !ok {
+				return nil, rdkutils.NewUnexpectedTypeError(conf, config.ConvertedAttributes)
+			}
+			return NewMotor(ctx, conf, logger)
+		},
+	}
+	registry.RegisterComponent(motor.Subtype, modelName, _motor)
+
+	config.RegisterComponentAttributeMapConverter(
+		motor.SubtypeName,
+		modelName,
+		func(attributes config.AttributeMap) (interface{}, error) {
+			var conf Config
+			decoder, err := mapstructure.NewDecoder(&mapstructure.DecoderConfig{TagName: "json", Squash: true, Result: &conf})
+			if err != nil {
+				return nil, err
+			}
+			if err := decoder.Decode(attributes); err != nil {
+				return nil, err
+			}
+			return &conf, nil
+		},
+		&Config{},
+	)
+}
+
+func newController(c *Config, logger golog.Logger) (*controller, error) {
+	if c.SerialAddress < 128 || c.SerialAddress > 135 {
```

Comment:
nit: this check is already done in the Validate function

Reply:
no validation gets called when a new config is submitted so you can't construct a sabertooth without validation step